src/TestableHttpClient/Response/SequencedResponse.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace TestableHttpClient.Response;
+
+internal class SequencedResponse : IResponse
+{
+    private readonly Queue<IResponse> responses;
+    public SequencedResponse(IEnumerable<IResponse> responses)
+    {
+        this.responses = new(responses ?? throw new ArgumentNullException(nameof(responses)));
+        if (this.responses.Count == 0)
+        {
+            throw new ArgumentException("Responses can't be empty.", nameof(responses));
+        }
+    }
+
+    public Task<HttpResponseMessage> GetResponseAsync(HttpRequestMessage requestMessage, CancellationToken cancellationToken)
+    {
+        var response = GetResponse();
+        return response.GetResponseAsync(requestMessage, cancellationToken);
+    }
+
+    private IResponse GetResponse()
+    {
+        if (responses.Count == 1)
```

Comment:
Consider adding the last response to a private field, so that we don’t have to count/peek every time.